### ISSUE TITLE
[ENHANCEMENT] Increase Replica Replenishment Wait Interval during upgrade

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#3671]
**Is your enhancement related to a problem? Please describe.**

The current Longhorn ["Replica Replenishment Wait Interval"](https://longhorn.io/docs/1.3.2/references/settings/#replica-replenishment-wait-interval) setting is just 10 minutes. If a replica is offline and online after 10 minutes, then the replica is discarded without being reused. Longhorn will find a new place to rebuild a new replica, which not only increases cluster I/O but the pre-drain phase time of the next upgrading node.

When upgrading and rebooting a node, the offline of a replica starts after the draining phase. The rest operations are:
- Upgrade RKE2
- Upgrade OS
- Reboot: modern machines need minutes to reboot, which generally means it's not easy to finish the whole process in 10 minutes.
- Node is back again.

**Describe the solution you'd like**

Increase the value to a reasonable value like 30 minutes and restore it to what the user sets after the upgrade is done.

**Describe alternatives you've considered**

Advise user to set the value before an upgrade.

**Additional context**

**Comment by innobead** [comment]
["Replica Replenishment Wait Interval"](https://longhorn.io/docs/1.3.2/references/settings/#replica-replenishment-wait-interval) can be used for intended node/cluster maintenance, so increasing it seems reasonable, but because this is a global setting, it would delay other volume replica rebuilding on other running nodes.

**Comment by starbops** [comment]
The plan is to increase the `replica-replenishment-wait-interval` LH setting from the default 600 to 1800 seconds at the beginning of each upgrade, and the setting will be configured back to the original value after the upgrade ends. The original value will be stored in the upgrade CR's specific annotation instead of creating a new field of the upgrade CRD.

**Comment by w13915984028** [comment]
@innobead Could LH also enhances to support volume level  "[Replica Replenishment Wait Interval"](https://longhorn.io/docs/1.3.2/references/settings/#replica-replenishment-wait-interval) setting ? thus Harvester can set the upgrade related volume with specific long wait value without touch the cluster level setting back and forth. thanks.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-f27ede00-head` (3/23). Close this issue.

### Result
During upgrade process enter into node upgrade stage
1. We can find `longhorn.io/replica-replenishment-wait-interval` annotation set in the upgrade CR
  ```
  node1:~ # kubectl -n harvester-system get upgrades hvst-upgrade-9j6px -o yaml
  apiVersion: harvesterhci.io/v1beta1
  kind: Upgrade
  metadata:
    annotations:
      harvesterhci.io/replica-replenishment-wait-interval: "600"
    creationTimestamp: "2023-03-24T08:09:46Z"
    finalizers:
    - wrangler.cattle.io/harvester-upgrade-controller
    generateName: hvst-upgrade-
    generation: 24
    labels:
      harvesterhci.io/latestUpgrade: "true"
      harvesterhci.io/upgradeState: UpgradingNodes
    name: hvst-upgrade-9j6px
    namespace: harvester-system
    resourceVersion: "135306"
    uid: 519c4a10-8274-4161-88a5-25e598e6dc50
  
  ```
2.  The Longhorn setting `replica-replenishment-wait-interval` is changed to 1800 (30 mins)

  ```
  node1:~ # kubectl -n longhorn-system get settings.longhorn.io replica-replenishment-wait-interval
  NAME                                  VALUE   AGE
  replica-replenishment-wait-interval   600     112m
  node1:~ # kubectl -n longhorn-system get settings.longhorn.io replica-replenishment-wait-interval
  NAME                                  VALUE   AGE
  replica-replenishment-wait-interval   1800    113m
  
  ```

After upgrade complete, the Longhorn setting `replica-replenishment-wait-interval` change back to default 600
  ```
  node1:~ # kubectl -n longhorn-system get settings.longhorn.io replica-replenishment-wit should be reverted back to the original value (normally 600, the default value)


ait-interval
  NAME                                  VALUE   AGE
  replica-replenishment-wait-interval   1800    119m
  node1:~ # kubectl -n longhorn-system get settings.longhorn.io replica-replenishment-wait-interval
  NAME                                  VALUE   AGE
  replica-replenishment-wait-interval   600     129m
  
  ```


### Test Information
* Test Environment: 3 nodes harvester on bare machines
* Harvester version: `master-f27ede00-head` (3/23)


### Verify Steps
1. Provision a three nodes Harvester cluster with master release
2. Upgrade the cluster with the same ISO image
3. Wait until upgrade proceed to node upgrade stage of any node
4. Execute command ` kubectl -n harvester-system get upgrades <upgrade-name> -o yaml` to check the replica-replenishment-wait-interval annotation exists
5. Execute command `kubectl -n longhorn-system get settings.longhorn.io replica-replenishment-wait-interval`
6. After the upgrade ends, check the Longhorn setting replica-replenishment-wait-interval, again 
7. Check it should be reverted back to the original value (normally 600, the default value)
